types/client.d.ts

```diff
@@ -70,7 +70,7 @@ export declare namespace Client {
     /** TODO */
     maxRedirections?: number;
     /** TODO */
-    connect?: buildConnector.BuildOptions | buildConnector.connector;
+    connect?: Partial<buildConnector.BuildOptions> | buildConnector.connector;
```

Comment:
`new Agent({ connect: { autoSelectFamily: true } }) is not actually considered valid currently because it doesn't specify required properties of `buildConnector.BuildOptions`. But the rest of the options come from elsewhere (hence why `connect` itself is not a required property) so you don't need to specify everything.